min-turnup/phase1/azure/lib/azure.jsonnet

```diff
@@ -0,0 +1,285 @@
+					name: names.storage_container,
+					container_access_type: "private"
+				}
+			},
+			azurerm_availability_set: {
+				as: {
+					resource_group_name: "${azurerm_resource_group.rg.name}",
+					name: names.availability_set,
+					location: "${azurerm_resource_group.rg.location}"
+				}
+			},
+			azurerm_virtual_network: {
+				vnet: {
+					resource_group_name: "${azurerm_resource_group.rg.name}",
+					location: "${azurerm_resource_group.rg.location}",
+					name: names.vnet,
+					address_space: ["10.0.0.0/8"],
+				}
+			},
+			azurerm_route_table: {
+				rt: {
+					resource_group_name: "${azurerm_resource_group.rg.name}",
+					location: "${azurerm_resource_group.rg.location}",
+					name: names.route_table,
+				}
+			},
+			azurerm_subnet: {
+				subnet: {
+					resource_group_name: "${azurerm_resource_group.rg.name}",
+					name: names.subnet,
+					virtual_network_name: "${azurerm_virtual_network.vnet.name}",
+					address_prefix: "10.0.0.0/16",
+					network_security_group_id: "${azurerm_network_security_group.sg.id}",
+					route_table_id: "${azurerm_route_table.rt.id}",
+				}
+			},
+			azurerm_network_security_group: {
+				sg: {
+					resource_group_name: "${azurerm_resource_group.rg.name}",
+					location: cfg.azure.location,
+					name: names.security_group,
+				},
+			},
+			azurerm_network_security_rule: {
+				[cfg.instance_prefix+"-master-ssh"]: {
+						name: "%(instance_prefix)s-master-ssh" % cfg,
+						priority: 100,
+						direction: "Inbound",
+						access: "Allow",
+						protocol: "Tcp",
+						source_port_range: "*",
+						destination_port_range: "22",
+						source_address_prefix: "*",
+						destination_address_prefix: "*",
+						resource_group_name: "${azurerm_resource_group.rg.name}",
+						network_security_group_name: "${azurerm_network_security_group.sg.name}",
+				},
+				[cfg.instance_prefix+"-master-ssl"]: {
+						name: "%(instance_prefix)s-master-ssl" % cfg,
+						priority: 110,
+						direction: "Inbound",
+						access: "Allow",
+						protocol: "Tcp",
+						source_port_range: "*",
+						destination_port_range: "443",
+						source_address_prefix: "*",
+						destination_address_prefix: "*",
+						resource_group_name: "${azurerm_resource_group.rg.name}",
+						network_security_group_name: "${azurerm_network_security_group.sg.name}",
+				},
+			},
+			azurerm_public_ip: {
+				pip: {
+					resource_group_name: "${azurerm_resource_group.rg.name}",
+					location: cfg.azure.location,
+					name: names.master_public_ip,
+					public_ip_address_allocation: "static",
+					provisioner:  [{
+						"local-exec": {
+							command: |||
+								cat <<EOF > ../../phase1b/crypto/san-extras
+								DNS.1 = kubernetes
+								DNS.2 = kubernetes.default
+								DNS.3 = kubernetes.default.svc
+								DNS.4 = kubernetes.default.svc.cluster.local
+								DNS.5 = names.master_vm
+								IP.1 = ${azurerm_public_ip.pip.ip_address}
+								IP.2 = 10.3.0.1
```

Comment:
+1

On Tue, 14 Jun 2016, 22:26 Mike Danese, <notifications@github.com> wrote:

> In min-turnup/phase1/azure/lib/azure.jsonnet
> <https://github.com/kubernetes/kube-deploy/pull/106#discussion_r67060401>:
>
> > +				pip: {
> > +					resource_group_name: "${azurerm_resource_group.rg.name}",
> > +					location: cfg.azure.location,
> > +					name: names.master_public_ip,
> > +					public_ip_address_allocation: "static",
> > +					provisioner:  [{
> > +						"local-exec": {
> > +							command: |||
> > +								cat <<EOF > ../../phase1b/crypto/san-extras
> > +								DNS.1 = kubernetes
> > +								DNS.2 = kubernetes.default
> > +								DNS.3 = kubernetes.default.svc
> > +								DNS.4 = kubernetes.default.svc.cluster.local
> > +								DNS.5 = names.master_vm
> > +								IP.1 = ${azurerm_public_ip.pip.ip_address}
> > +								IP.2 = 10.3.0.1
>
> We should probably make node cidr, service cidr, pod cidr configurable
> ASAP.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kube-deploy/pull/106/files/f56f5e25dce862b13c240daede4900fe2e9c680e#r67060401>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAPWS4rcn0N4jZ2HvDAL4Jnapaf7vQnvks5qLxx7gaJpZM4IzsbA>
> .
>